### PR TITLE
Feature/readme n docstrings

### PR DESCRIPTION
**Pull request by kouyoumart**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,31 +1,103 @@
...
-    - lib: this is where you store the main libraries used within your project. 
-    - data: Separated in 3 folder to start: Raw data, intermediate, and processed.
-    - doc: Sphinx template to generate code documentation
-    - references: all the written documentation (functional, features) that is not sphinx generated 
+The app's aim is to: 
+- help people find arXiv articles linked to a subject they are interested in
+- illustrate how the subject of interest has evolved throughout history
 
-### Requirements
+It takes as input a user's query - a sentence describing what type of arXiv papers a person is interested in searching.
```
Comment (Youssef-Moutaouakil) [inline README.md]: It takes as input a user's query - a sentence describing what type of arXiv papers a person is interested in searching - along with a set of filters on the year, the categories, and the number of similar papers to be retrieved.

File: README.md [hunk 2]
```diff
@@ -1,31 +1,103 @@
...
+Using **RedisSearch** capabilities, a set of similar articles, linked to the query, is obtained. 
 
-Requirements are often a pain, between the one that you really use, and all the dependencies.
-This repo advise you to use [Pip-tools](https://github.com/jazzband/pip-tools).
+Three output sections are then presented: 
 
-To do so, put your requirements in the requirements.in, then run 
+1. Topic trend & future projection
+2. Topic evolution
+3. Papers overview
```
Comment (Youssef-Moutaouakil) [inline README.md:35]: 1. Topic trend & future projection
2. Topic evolution and paper interconnection through a citations graph
3. Reading list recommender - based on the papers' similarity scores and node degree in the citation graph
4. Papers overview

File: README.md [hunk 3]
```diff
@@ -1,31 +1,103 @@
...
 
-    The folder name should be self sufficient, however some extra details:
-    - bin: This is the folder where you store your executable, that could be main python scripts, or bash ones.
-    - lib: this is where you store the main libraries used within your project. 
-    - data: Separated in 3 folder to start: Raw data, intermediate, and processed.
-    - doc: Sphinx template to generate code documentation
-    - references: all the written documentation (functional, features) that is not sphinx generated 
+The app's aim is to: 
+- help people find arXiv articles linked to a subject they are interested in
+- illustrate how the subject of interest has evolved throughout history
```
Comment (Youssef-Moutaouakil) [inline README.md]: Je propose cette version plus détaillée: 
The app's vision is to combine vector search and co-citations graph structures in order to:
- Help the user find arXiv articles linked to a subject he's interested in
- Provide the user with a view of a topic's trend, based on the number of related publications throughout time, along with a trend prediction for the next two years
- Illustrate how the subject of interest has evolved throughout history, with an arc diagram highlighting the founding / most influential papers and co-citation relationships between papers
- Recommend a curated reading list, based on a combination of vector similarity score and paper node degree in the citations graph

File: README.md [hunk 4]
```diff
@@ -1,31 +1,103 @@
-# Artefact's Boilerplate for  DataScience Python projects 
+# AreYouRedis - Vector Search Engineering Hackathon 
 
-This repository is a boilerplate repository designed to be used when starting a new project to help kickstart things easily.
+This repository is the AreYouRedis team's hackathon submission.
```
Comment (Youssef-Moutaouakil) [inline README.md]: Je sais pas si cette phrase est utile ?
Sinon on pourrait dire:
This repository contains the code base built by the AreYouRedis team to create  the "Darwinian Paper Explorer" app.

File: README.md (hunk 3 above)
Comment (kouyoumart) [reply README.md]: top, je change

**Pull request merged**